valohai_yaml/objs/deployment.py

```diff
@@ -0,0 +1,41 @@
+from typing import Optional
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.types import SerializedDict
+
+
+class DeploymentDefaults(Item):
+    """Represents the default values to use *if* matching deployment does not exist in the project context."""
+
+    def __init__(
+        self,
+        *,
+        target: str,
+        allow_cors: Optional[bool] = None,
+        allow_configuration_snippets: Optional[bool] = None,
+    ) -> None:
+        self.target = target
+        self.allow_cors = allow_cors
+        self.allow_configuration_snippets = allow_configuration_snippets
+
+
+class Deployment(Item):
+    """Represents a deployment in a project under where deployment versions and endpoints are housed."""
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        defaults: Optional[DeploymentDefaults] = None,
```

Comment:
A `Deployment` without a `defaults` is pretty useless though, right?

Reply:
yes! does basically nothing except if you want to use it to document expected deployments the project needs

like I mentioned in the other comment, the `deployment` definition could be used as a baseline when creating deployments on the web UI so it might be helpful to suggest names for deployments, but that is not inside the current scope I'm working on

Reply:
this I left as-is, let me know if you want changes